components/brave_vpn/common/wireguard/win/storage_utils.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/common/wireguard/win/storage_utils.h"
+
+#include "base/logging.h"
+#include "base/win/registry.h"
+#include "brave/components/brave_vpn/common/wireguard/win/service_constants.h"
+
+namespace brave_vpn {
+
+namespace wireguard {
+
+namespace {
+constexpr wchar_t kBraveWireguardConfigKeyName[] = L"ConfigPath";
+constexpr wchar_t kBraveWireguardEnableTrayIconKeyName[] = L"EnableTrayIcon";
+}  // namespace
+
+bool IsVPNTrayIconEnabled() {
+  base::win::RegKey storage;
+  if (storage.Open(
+          HKEY_CURRENT_USER,
+          brave_vpn::GetBraveVpnWireguardServiceRegistryStoragePath().c_str(),
+          KEY_QUERY_VALUE) != ERROR_SUCCESS) {
+    return true;
+  }
+  DWORD value = 1;
+  if (storage.ReadValueDW(kBraveWireguardEnableTrayIconKeyName, &value) !=
+      ERROR_SUCCESS) {
+    return true;
+  }
```

Comment:
We return `true` when reg operation failed. Is this intended?

Reply:
yes, it is disabled only if we have zero set